### PR TITLE
[WIP] Merge PRs

### PR DESCRIPTION
**Pull request by rnystrom**
- Checks when setup
- Individual check status
- Mergeability (conflict, etc)
- Button w/ different types

Still TODO:

- Interactions
- Create a configure store to set the default merge type per repo
- Send merge API request

### CODE REVIEW

**Comment by rnystrom** [comment]
img_1318

img_1315

**Comment by rnystrom** [comment]
cc @orta 😉

**Comment by orta** [comment]
This is very relevant to my interests

**Comment by BasThomas** [comment]
Slick!

File: Classes/Issues/Merge/IssueMergeButtonCell.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        buttonBackgroundView.snp.makeConstraints { make in
+            make.edges.equalToSuperview().inset(UIEdgeInsets(
+                top: Styles.Sizes.rowSpacing/2,
+                left: Styles.Sizes.commentGutter,
+                bottom: Styles.Sizes.rowSpacing,
+                right: Styles.Sizes.commentGutter
+            ))
+        }
+
+        optionButton.setImage(UIImage(named: "chevron-down")?.withRenderingMode(.alwaysTemplate), for: .normal)
```
Comment (BasThomas) [inline Classes/Issues/Merge/IssueMergeButtonCell.swift]: Can we add an `accessibilityLabel` to this guy? :)

File: Classes/Issues/Merge/IssueMergeContextCell.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        contentView.addSubview(avatarView)
+        contentView.addSubview(iconView)
+        contentView.addSubview(titleLabel)
+        contentView.addSubview(descriptionLabel)
+
+        avatarView.clipsToBounds = true
```
Comment (BasThomas) [inline Classes/Issues/Merge/IssueMergeContextCell.swift]: Can we add the ignore invert colors on this view?

```swift
if #available(iOS 11, *) {
  avatarView.accessibilityIgnoresInvertColors = true
}
```

File: Classes/Issues/Merge/IssueMergeContextCell.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            iconName = "check-small"
+            iconColor = Styles.Colors.Green.medium.color
+        case .error, .failure:
+            iconName = "x-small"
+            iconColor = Styles.Colors.Red.medium.color
+        case .pending:
+            iconName = "primitive-dot"
+            iconColor = Styles.Colors.Yellow.medium.color
+        }
+        iconView.image = UIImage(named: iconName)?.withRenderingMode(.alwaysTemplate)
```
Comment (BasThomas) [inline Classes/Issues/Merge/IssueMergeContextCell.swift]: A11y?

File: Classes/Issues/Merge/IssueMergeSectionController.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    func sectionController(
+        _ sectionController: ListBindingSectionController<ListDiffable>,
+        viewModelsFor object: Any
+        ) -> [ListDiffable] {
+        guard let object = self.object else { return [] }
+
+        var viewModels = [ListDiffable]()
+
+        if object.contexts.count > 0 {
+            let success = object.contexts.reduce(true, { $0 && $1.state == .success })
```
Comment (BasThomas) [inline Classes/Issues/Merge/IssueMergeSectionController.swift]: Nice!

File: Classes/Issues/Merge/IssueMergeSummaryCell.swift [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            iconName = "merge-check"
+        case .failure:
+            imageViewBackground = Styles.Colors.Red.medium.color
+            iconName = "merge-x"
+        case .warning:
+            imageViewBackground = Styles.Colors.Gray.medium.color
+            iconName = "merge-alert"
+        }
+        imageView.backgroundColor = imageViewBackground
+        imageView.image = UIImage(named: iconName)?.withRenderingMode(.alwaysTemplate)
```
Comment (BasThomas) [inline Classes/Issues/Merge/IssueMergeSummaryCell.swift]: A11y?

File: Classes/Issues/Merge/IssueMergeButtonCell.swift (hunk 1 above)
Comment (rnystrom) [reply Classes/Issues/Merge/IssueMergeButtonCell.swift]: Yes! I’m gonna refactor this whole button too. Good eye.

File: Classes/Issues/Merge/IssueMergeContextCell.swift (hunk 2 above)
Comment (rnystrom) [reply Classes/Issues/Merge/IssueMergeContextCell.swift]: Sounds good

**Comment by rnystrom** [comment]
Will followup w/ a11y after we get some feedback

**Pull request merged**